client/src/pages/private/PSITable.js

```diff
@@ -0,0 +1,235 @@
+        'Content-type': 'application/json',
+      },
+      body: JSON.stringify(psi),
+    });
+
+    if (response.ok) {
+      setActiveModalForm(null);
+      fetchPSIs();
+    } else {
+      const error = await response.json();
+      if (error.status) {
+        if (error.status === '23505') {
+          setActiveModalForm(null);
+        }
+        openToast({
+          status: ToastStatus.Error,
+          message: error.error || response.error || response.statusText || 'Server error',
+        });
+      }
+    }
+  };
+  useEffect(() => {
+    setHealthAuthorities(
+      roles.includes('superuser') || roles.includes('ministry_of_health')
+        ? Object.values(regionLabelsMap)
+        : roles.map((loc) => regionLabelsMap[loc]).filter(Boolean)
+    );
+  }, [roles]);
+
+  const sort = (array) => _orderBy(array, [orderBy, 'operatorName'], [order]);
+
+  useEffect(() => {
+    fetchPSIs();
+    // This fetch PSIs is a dependency of this function. This needs to be reworked, but it is outside of the scope of the ticket
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [history, location]);
+
+  const defaultOnClose = () => {
+    setActiveModalForm(null);
+  };
+  return (
+    <>
+      <Dialog
+        title={`Create New Institute`}
+        open={activeModalForm != null}
+        onClose={defaultOnClose}
+      >
+        {activeModalForm === 'new-psi' && (
+          <NewPSIForm
+            initialValues={{
+              instituteName: '',
+              healthAuthority: '',
+              postalCode: '',
+            }}
+            validationSchema={CreatePSISchema}
+            onSubmit={(values) => {
+              handlePSICreate({
+                instituteName: values.instituteName,
+                healthAuthority: values.healthAuthority,
+                postalCode: values.postalCode,
+              });
+            }}
+            onClose={defaultOnClose}
+          />
+        )}
+      </Dialog>
+      <Grid
+        container
+        alignContent='flex-start'
+        justify='flex-start'
+        alignItems='center'
+        direction='column'
+      >
+        <Grid
+          container
+          alignContent='flex-start'
+          justify='flex-start'
+          alignItems='center'
+          direction='row'
+        >
+          <Grid item>
+            <Box pl={2} pr={2} pt={1}>
+              <Typography variant='body1' gutterBottom>
+                Filter:
+              </Typography>
+            </Box>
+          </Grid>
+          <Grid item>
+            <Box minWidth={180}>
+              <TableFilter
+                onFilter={(filteredRows) => setRows(filteredRows)}
+                values={healthAuthorities}
+                rows={fetchedRows}
+                label='Health Authority'
+                filterField='health_authority'
+              />
+            </Box>
+          </Grid>
+          <CheckPermissions roles={roles} permittedRoles={['ministry_of_health']}>
```

Comment:
Isn't this whole page only accessible by MoH?

Reply:
For now, we're adding Health Authorities at a later date.